### PR TITLE
GCS: Solves Logging crash when existing. Fixes #746

### LINKED ISSUE #746: GCS: Quitting when logging causes a crash
`File closed
QThread: Destroyed while thread is still running
Dump file was written to: /var/folders/kb/g2m5j1f576l3z7lk3ypyq6hr0000gn/T/dr_mpl_connectfaster_20160304_1a3e402c/73C00019-BA51-4767-82D0-12D5869BD092.dmp`

or sometimes it doesn't crash:

`File closed
QThread: Destroyed while thread is still running
QWaitCondition::wakeAll(): mutex lock failure: Invalid argument
QWaitCondition::wakeAll(): cv broadcast failure: Invalid argument
QWaitCondition::wakeAll(): mutex unlock failure: Invalid argument`

### PR DESCRIPTION
**Pull request by jpbarraca**
Let the thread exit and wait for it during shutdown.

edit by mpl: fixes #746 -- github is picky that this has to be in this top comment block

### CODE REVIEW

File: ground/gcs/src/plugins/logging/loggingplugin.cpp [hunk 1]
```diff
@@ -485,7 +484,16 @@ void LoggingPlugin::extensionsInitialized()
...
 void LoggingPlugin::shutdown()
 {
-    // Do nothing
+    if(state == LOGGING) {
+        stopLogging();
+
+        loggingThread->wait();
+    }
+
+    if (loggingThread == NULL) {
```
Comment (mlyle) [inline ground/gcs/src/plugins/logging/loggingplugin.cpp]: This should be !=, right?

**Comment by mlyle** [comment]
Thanks for fixing this :)

File: ground/gcs/src/plugins/logging/loggingplugin.cpp (hunk 1 above)
Comment (tracernz) [inline ground/gcs/src/plugins/logging/loggingplugin.cpp]: Should maybe enclose the previous if too?

File: ground/gcs/src/plugins/logging/loggingplugin.cpp [hunk 2]
```diff
@@ -485,7 +484,16 @@ void LoggingPlugin::extensionsInitialized()
 
 void LoggingPlugin::shutdown()
 {
-    // Do nothing
+    if(state == LOGGING) {
```
Comment (tracernz) [inline ground/gcs/src/plugins/logging/loggingplugin.cpp]: Missing space between if and (.

File: ground/gcs/src/plugins/logging/loggingplugin.cpp (hunk 1 above)
Comment (jpbarraca) [inline ground/gcs/src/plugins/logging/loggingplugin.cpp]: damn... I will correct it asap :/

**Comment by mlyle** [comment]
Works great.

**Pull request merged**